### ISSUE TITLE
Address comment about allocations in `MergedNamespaceSymbol.GetTypeMembers`

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#68943]
https://github.com/dotnet/roslyn/blob/33496991002bcf1093028e82671fc5c0de2aa4bf/src/Compilers/CSharp/Portable/Symbols/MergedNamespaceSymbol.cs#L244-L247

**Comment by jcouv** [comment]
@Youssef1313 I see that you filed multiple perf related suggestions recently. I don't really know how to triage them without data showing a hot spot. I'll mark them as Help Wanted and contributions are welcome if we can show a positive impact on perf.

**Comment by Youssef1313** [comment]
Yeah, have been capturing lots of traces for a while, and also going through the code by eyes.

This specific one was when going through the code
